src/foam/dao/PostgresDAO.java

```diff
@@ -0,0 +1,175 @@
+package foam.dao;
+
+import foam.core.*;
+
+import java.sql.*;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+public class PostgresDAO extends ProxyDAO {
+
+    public PostgresDAO(DAO delegate, String host, String port, String dbName, String username, String password) {
+        setDelegate(delegate);
+
+        if (dbName == null || username == null) {
+            throw new IllegalArgumentException("Illegal arguments");
+        }
+
+        host = (host != null) ? host : "localhost";
+        port = (port != null) ? port : "5432";
+
+        PgConnectionPool.setup(host, port, dbName, username, password);
+    }
+
+    @Override
+    public FObject remove(FObject o) {
+
+        try {
+
+            SQLData data = new SQLData(o);
+            Connection c = PgConnectionPool.getConnection();
+
+            PreparedStatement smt = c.prepareStatement(data.createDeleteStatement());
+
+            smt.setLong(1, ((Long) o.getProperty("id")));
+
+            int removed = smt.executeUpdate();
+            if (removed == 0) {
+                throw new SQLException("Error while removing.");
+            }
+            return o;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return null;
+    }
+
+    /**
+     * maps a database result row to an FObject
+     *
+     * @param row
+     * @return FObject
+     * @throws SQLException
+     */
+    private FObject createFObject(ResultSet row) throws Exception {
+
+        if (getOf() == null) {
+            throw new Exception("`Of` is not set");
+        }
+
+        List<PropertyInfo> props = getOf().getAxioms();
+        FObject result = (FObject) getOf().getObjClass().newInstance();
+
+        // set fields
+        for(int i = 0; i < props.size(); i++ ) {
+            String propName = props.get(i).getName();
+            Object value = row.getObject(i + 1);
+            result.setProperty(propName, value);
```

Comment:
If you call the set method on props.get(i) it will avoid looking up the property by name and be faster.